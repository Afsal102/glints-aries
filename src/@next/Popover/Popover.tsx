import React from 'react';
import {
  Popover as PolarisPopover,
  PopoverProps as PolarisPopoverProps,
} from 'polaris-glints';
import { StyledPopover } from './PopoverStyle';
export type PopoverProps = PolarisPopoverProps;
const PopoverComponent = (props: PopoverProps) => {
  return (
    <>
      <StyledPopover />
      <PolarisPopover {...props} />
    </>
  );
};

/** Using Popover from polaris-glints repository internally
 * Please add `<div id="glints-portal-container"></div>` as a sibling element of where your React app is rendered.
 * This is a temporary workaround, as React Portal need this as the anchor */
export const Popover = Object.assign(PopoverComponent, {
  Pane: PolarisPopover.Pane,
  Section: PolarisPopover.Section,
});
