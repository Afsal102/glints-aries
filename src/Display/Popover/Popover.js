import React, { Component } from 'react';
import {
  ChildHolder, ButtonContainer, PopOverContent, CursorPointer,
} from '../../Style/Display/PopoverStyle';

class Popover extends Component <Props, State> {
  static defaultProps = {
    margin: '',
  }

  constructor() {
    super();
    this.state = {
      isOpen: false,
    };
  }

  handleOnClick = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  handleClickOutside = () => {
    this.setState({ isOpen: false });
  }

  renderChildren = (children) => {
    const { isOpen } = this.state;
    const { margin, content } = this.props;
    return (
      <ChildHolder onClick={this.handleOnClick} margin={margin}>
        <CursorPointer>{children}</CursorPointer>
        <If condition={isOpen}>
          <PopOverContent>
            {content}
          </PopOverContent>
        </If>
      </ChildHolder>
    );
  }

  render() {
    const { children } = this.props;
    return (
      <ButtonContainer
        tabIndex="0"
        onBlur={this.handleClickOutside}
      >
        {this.renderChildren(children)}
      </ButtonContainer>
    );
  }
}


type Props = {
  children: React$Node,
  content: React$Node,
};

type State = {
  isOpen: boolean,
};

export default Popover;
