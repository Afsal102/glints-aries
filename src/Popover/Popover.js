import React, { Component } from 'react';
import { ChildHolder, ButtonContainer, PopOverContent } from '../Style/PopoverStyle';

class Popover extends Component <Props, State> {
  static defaultProps = {
    margin: '',
  }

  constructor() {
    super();
    this.state = {
      isOpen: true,
    };
  }

  componentDidMount() {
    this.handleOnClick();
  }

  handleOnClick = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  handleClickOutside = () => {
    this.setState({ isOpen: false });
  }

  renderChildren = (children) => {
    const { margin } = this.props;
    return (
      <ChildHolder onClick={this.handleOnClick} margin={margin}>
        {children}
      </ChildHolder>
    );
  }

  render() {
    const { isOpen } = this.state;
    const { content, children } = this.props;

    return (
      <ButtonContainer
        tabIndex="0"
        onBlur={this.handleClickOutside}
      >
        {this.renderChildren(children)}
        <If condition={isOpen}>
          <PopOverContent>
            {content}
          </PopOverContent>
        </If>
      </ButtonContainer>
    );
  }
}


type Props = {
  children: React$Node,
  content: React$Node,
};

type State = {
  isOpen: boolean,
};

export default Popover;
