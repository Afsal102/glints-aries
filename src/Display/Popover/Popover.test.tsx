import * as React from 'react';

import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import Popover from './Popover';

const PopoverContent = () => (
  <div style={{ backgroundColor: 'white' }}> Popover Content </div>
);

describe('<Popover /> prop className', () => {
  const matchSnapshot = (className: string) => {
    test(`should match snapshot when class name ${className} is passed`, () => {
      const { asFragment } = render(
        <Popover content={<PopoverContent />} className={className}>
          <button key="button"> Open Popover </button>
        </Popover>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  };

  ['test', undefined].forEach(matchSnapshot);

  test('content prop should be null when popover is rendered', () => {
    const { queryByText } = render(
      <Popover content={<PopoverContent />}>
        <button key="button"> Open Popover </button>
      </Popover>
    );

    const contentText = queryByText('Popover Content');
    expect(contentText).toBeNull();
  });
});

describe('<Popover /> prop children', () => {
  test(`should render 1 Popover children`, () => {
    render(
      <Popover content={<PopoverContent />}>
        <button key="button"> Open Popover </button>
      </Popover>
    );

    const childrenCount = 1;
    const item = document.querySelectorAll('.popover-children');
    expect(item).toHaveLength(childrenCount);
  });
});

describe('<Popover/> click event to show/close content', () => {
  const onPopoverClick = jest.fn();

  test('content prop should be visible when popover clicked once', () => {
    const { queryByTestId, queryByText } = render(
      <Popover content={<PopoverContent />}>
        <button key="button"> Open Popover </button>
      </Popover>
    );

    const contentTextBeforeClick = queryByText('Popover Content');
    expect(contentTextBeforeClick).toBeNull();
    expect(onPopoverClick).not.toHaveBeenCalled();

    const children = queryByTestId('children');
    fireEvent.click(children);

    const contentTextAfterClick = queryByText('Popover Content');
    expect(contentTextAfterClick).toBeVisible();
  });

  test('content prop should be closed when the component is clicked twice', () => {
    const { queryByTestId, queryByText } = render(
      <Popover content={<PopoverContent />}>
        <button key="button"> Open Popover </button>
      </Popover>
    );

    expect(queryByText('Popover Content')).toBeNull();
    const children = queryByTestId('children');
    fireEvent.click(children);

    expect(queryByText('Popover Content')).toBeVisible();
    fireEvent.click(children);
    expect(queryByText('Popover Content')).toBeNull();
  });
});
