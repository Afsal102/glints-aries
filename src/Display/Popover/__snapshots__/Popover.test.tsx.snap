// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> prop className should match snapshot when class name test is passed 1`] = `
<DocumentFragment>
  .c5 {
  vertical-align: middle;
}

.c0 {
  position: relative;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  cursor: pointer;
}

.c1 {
  position: relative;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 1em;
}

.c3 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5em;
}

<div
    aria-busy="false"
    aria-haspopup="true"
    class="c0 aries-popover test"
    tabindex="0"
  >
    <div
      class="c1 popover-children"
      data-testid="children"
    >
      <button>
         Open Popover 
      </button>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            class="c5"
            data-testid="icon-svg"
            fill="black"
            height="1em"
            viewBox="0 0 100 100"
            width="1em"
          >
            <polygon
              points="11.7984725 19 49.9935275 57.3950291 88.1885825 19 99.987055 30.7984725 49.9935275 80.792 0 30.7984725"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Popover /> prop className should match snapshot when class name undefined is passed 1`] = `
<DocumentFragment>
  .c5 {
  vertical-align: middle;
}

.c0 {
  position: relative;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  cursor: pointer;
}

.c1 {
  position: relative;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 1em;
}

.c3 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5em;
}

<div
    aria-busy="false"
    aria-haspopup="true"
    class="c0 aries-popover"
    tabindex="0"
  >
    <div
      class="c1 popover-children"
      data-testid="children"
    >
      <button>
         Open Popover 
      </button>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <svg
            class="c5"
            data-testid="icon-svg"
            fill="black"
            height="1em"
            viewBox="0 0 100 100"
            width="1em"
          >
            <polygon
              points="11.7984725 19 49.9935275 57.3950291 88.1885825 19 99.987055 30.7984725 49.9935275 80.792 0 30.7984725"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
